pkg/csi/service/common/vsphereutil.go

```diff
@@ -605,10 +605,29 @@ func ExpandVolumeUtil(ctx context.Context, manager *Manager, volumeID string, ca
 		log.Infof("Successfully expanded volume for volumeid %q to new size %d Mb.", volumeID, capacityInMb)
 		return "", nil
 	} else {
-		expansionRequired, err := isExpansionRequired(ctx, volumeID, capacityInMb, manager, useAsyncQueryVolume)
+		var expansionRequired bool
+		vc, err := GetVCenter(ctx, manager)
 		if err != nil {
+			log.Errorf("failed to get vcenter. err=%v", err)
 			return csifault.CSIInternalFault, err
 		}
+		isvSphere8AndAbove, err := IsvSphere8AndAbove(ctx, vc.Client.ServiceContent.About)
+		if err != nil {
+			return "", logger.LogNewErrorf(log,
```

Comment:
Do you think returning an error is important here? Can we skip and call QueryVolume anyways?

Reply:
+1